src/redisearch_rs/inverted_index/src/index_result.rs

```diff
@@ -828,6 +867,34 @@ impl<'index> RSIndexResult<'index> {
         }
     }
 
+    /// Get the aggregate result associated with this record
+    /// **without checking the discriminant**.
+    ///
+    /// # Safety
+    ///
+    /// 1. `Self::is_aggregate` must return `true` for `self`.
+    pub unsafe fn as_aggregate_unchecked(&self) -> Option<&RSAggregateResult<'index>> {
+        debug_assert!(
+            self.is_aggregate(),
+            "Invariant violation: `as_aggregate_unchecked` was invoked on an `IndexResult` \
+            instance that didn't actually contain an aggregate! It was a {}",
+            self.data.kind()
+        );
+        match &self.data {
+            RSResultData::Union(agg)
+            | RSResultData::Intersection(agg)
+            | RSResultData::HybridMetric(agg) => Some(agg),
+            RSResultData::Term(_)
+            | RSResultData::Virtual
+            | RSResultData::Numeric(_)
+            | RSResultData::Metric(_) => {
+                // SAFETY:
+                // - Thanks to safety precondition 1., we'll never reach this statement.
+                unsafe { std::hint::unreachable_unchecked() }
+            }
```

Comment:
If I understand this correctly, this `match` check will be compiled away because of the `unreachable_unchecked` hint right?